app/controllers/candidates/registrations/personal_informations_controller.rb

```diff
@@ -7,11 +7,21 @@ def new
 
       def create
         @personal_information = PersonalInformation.new personal_information_params
-        if @personal_information.valid?
-          persist @personal_information
+        render(:new) && return unless @personal_information.valid?
+
+        persist @personal_information
+
+        if candidate_signed_in?
           redirect_to new_candidates_school_registrations_contact_information_path
         else
-          render :new
+          token = @personal_information.create_signin_token(gitis_crm)
```

Comment:
does this need feature flagging in case we deploy but don't switch phase?

Reply:
Yeh - the whole thing needs feature flagging - thats not in at present but I want it in before I merge the epic branch into `master`. More broadly there's some questions about the release steps to be answered first

Reply:
I've created a separate story to track this - SE-1257

Theres a few places which will need feature flagging before the epic can get merged to master but I want to discuss how this roles out first anyway